flowkit/config/dependency.go

```diff
@@ -0,0 +1,79 @@
+/*
+ * Flow CLI
+ *
+ * Copyright 2019 Dapper Labs, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package config
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/onflow/flow-go-sdk"
+)
+
+func ParseRemoteSourceString(s string) (network, address, contractName string, err error) {
+	parts := strings.Split(s, "://")
+	if len(parts) != 2 {
+		return "", "", "", fmt.Errorf("invalid dependency source format")
+	}
+	network = parts[0]
+
+	subParts := strings.Split(parts[1], ".")
+	if len(subParts) != 2 {
+		return "", "", "", fmt.Errorf("invalid dependency source format")
+	}
+	address = subParts[0]
+	contractName = subParts[1]
+
+	return network, address, contractName, nil
+}
+
+type RemoteSource struct {
+	NetworkName  string
+	Address      flow.Address
+	ContractName string
+}
+
+type Dependency struct {
+	Name         string
+	RemoteSource RemoteSource
+	Version      string
```

Comment:
I noticed the hash is used as version, maybe we can call it `CodeHash` for clarity? What do you think?

Reply:
I struggled on naming this one. Finding the right balance of being understandable and being specific about what it was was difficult. I don't really have an opinion. Other options are just `Hash`, `CheckSum`, `Id`